sphinx/builders/linkcheck.py

```diff
@@ -111,99 +111,97 @@ def process_result(self, result: CheckResult) -> None:
         }
         self.write_linkstat(linkstat)
 
-        if result.status == _Status.UNCHECKED:
-            return
-        if result.status == _Status.WORKING and result.message == 'old':
-            return
```

Comment:
Commenting for future historians: the `old` message tag here was a way that the `linkcheck` component would communicate to itself that a hyperlink had already been checked and confirmed as working ('good').

That code was removed in 43517219b7813c0c48d8a0c69656a8de7ab1df91, and I believe that the URI deduplication that was removed there is no longer necessary, because `linkcheck` nowadays collects hyperlinks in a Python `dict`, `CheckExternalLinksBuilder.hyperlinks`, keyed by the URI; so duplicates should not occur anyway.